xun/functions/function.py

```diff
@@ -82,7 +82,7 @@ def sha256(desc, dependencies):
         str
             Hex digest of function hash
         """
-        content = ast.dump(desc.ast).encode()
+        content = desc.src.encode1()
```

Comment:
You, as you know, had a typo
```suggestion
        content = desc.src.encode()
```